### PR TITLE
fix: explore csv export

### PR DESCRIPTION
**Pull request by Chrischuck**
Closes: #4628 

### Description:

Looks like `projectUuid` was undefined. Is there ever a case where it is not undefined?

### CODE REVIEW

File: packages/common/src/types/savedCharts.ts [hunk 1]
```diff
@@ -233,7 +233,6 @@ export type CreateSavedChartVersion = Omit<
     | 'uuid'
     | 'name'
     | 'updatedAt'
-    | 'projectUuid'
```
Comment (IrakliJani) [inline packages/common/src/types/savedCharts.ts:236]: can we revert this change? `CreateSavedChartVersion` is referenced in many different places

File: packages/frontend/src/providers/ExplorerProvider.tsx [hunk 2]
```diff
@@ -211,6 +211,7 @@ const defaultState: ExplorerReduceState = {
     expandedSections: [ExplorerSection.RESULTS],
     unsavedChartVersion: {
         tableName: '',
+        projectUuid: '',
```
Comment (IrakliJani) [inline packages/frontend/src/providers/ExplorerProvider.tsx:214]: we should revert this as well

File: packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx [hunk 3]
```diff
@@ -88,9 +88,9 @@ const VisualizationCard: FC = memo(() => {
     }
 
     const getCsvLink = async (csvLimit: number | null, onlyRaw: boolean) => {
-        if (explore?.name && unsavedChartVersion?.metricQuery && projectUuid) {
+        if (explore?.name && unsavedChartVersion?.metricQuery) {
             const csvResponse = await downloadCsv({
-                projectUuid,
+                projectUuid: projectUuid || unsavedChartVersion.projectUuid,
```
Comment (IrakliJani) [inline packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx:93]: you can get a fallback projectUuid from the URI by using
```
const { projectUuid } = useParams<{ projectUuid: string }>();
```
I would add this chunk of code to `Explore` component rather than `VisualizationCard` and pass it down as a prop.

then you can use `savedChart.projectUuid || projectUuid` in here

LMK if you have questions!

**Review by IrakliJani — COMMENTED** [review]
nice work, commented how to get fallback projectUuid param

File: packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx [hunk 4]
```diff
@@ -24,134 +24,141 @@ export type EchartsClickEvent = {
...
-    );
-    const toggleSection = useCallback(
-        () => toggleExpandedSection(ExplorerSection.VISUALIZATION),
-        [toggleExpandedSection],
-    );
-    const projectUuid = useExplorerContext(
-        (context) => context.state.savedChart?.projectUuid,
-    );
+const VisualizationCard: FC<{ projectUuid?: string }> = memo(
+    ({ projectUuid: fallBackUUid }) => {
```
Comment (Chrischuck) [inline packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx:28]: I think this formatting made the diff way bigger than it actually is...

**Review by IrakliJani — APPROVED** [review]
works as expected. great job 🙌
